graphql-lint/src/lib.rs

```diff
@@ -0,0 +1,654 @@
+                    ));
+                }
+            }
+            "Mutation" => {
+                for prefix in ["mutation", "put", "post", "patch"] {
+                    if field_name.starts_with(prefix) {
+                        self.diagnostics.push((
+                            format!("field '{field_name}' on type Mutation has a forbidden prefix: '{prefix}'"),
+                            Severity::Warning,
+                        ));
+                        break;
+                    }
+                }
+                if field_name.ends_with("Mutation") {
+                    self.diagnostics.push((
+                        format!("field '{field_name}' on type Mutation has a forbidden suffix: 'Mutation'"),
+                        Severity::Warning,
+                    ));
+                }
+            }
+            "Subscription" => {
+                if field_name.starts_with("subscription") {
+                    self.diagnostics.push((
+                        format!("field '{field_name}' on type Subscription has a forbidden prefix: 'subscription'"),
+                        Severity::Warning,
+                    ));
+                }
+                if field_name.ends_with("Subscription") {
+                    self.diagnostics.push((
+                        format!("field '{field_name}' on type Subscription has a forbidden suffix: 'Subscription'"),
+                        Severity::Warning,
+                    ));
+                }
+            }
+            _ => {}
+        }
+    }
+
+    pub fn visit_directive(&mut self, directive: &DirectiveDefinition<'a>) {
+        if let CaseMatch::Incorrect { current, fix } = Self::case_check(directive.name(), Case::Camel) {
+            self.diagnostics.push((
+                format!("directive '{current}' should be renamed to '{fix}'"),
+                Severity::Warning,
+            ));
+        }
+    }
+
+    pub fn visit_directive_usage(&mut self, parent: &TypeDefinition<'a>, directive: &Directive<'a>) {
+        if directive.name() == "deprecated" && !directive.arguments().any(|argument| argument.name() == "reason") {
+            self.diagnostics.push((
+                format!(
+                    "usage of directive 'deprecated' on {} '{}' does not populate the 'reason' argument",
+                    Self::type_definition_display(parent),
+                    parent.name()
+                ),
+                Severity::Warning,
+            ));
+        }
+    }
+
+    pub fn visit_directive_usage_field(
+        &mut self,
+        parent_type: &TypeDefinition<'a>,
+        parent_field: &FieldDefinition<'a>,
+        directive: &Directive<'a>,
+    ) {
+        if directive.name() == "deprecated" && !directive.arguments().any(|argument| argument.name() == "reason") {
+            self.diagnostics.push((
+                format!(
+                    "usage of directive 'deprecated' on field '{}' on {} '{}' does not populate the 'reason' argument",
```

Comment:
Same comment about {}.{} here (not offering a suggestion though, because I'm not exactly sure what the three `{}`s are 😓